app/javascript/add-another.js

```diff
@@ -63,6 +63,10 @@ const setUpRemoveButton = (section) => {
   if (!removeButton) {
     return;
   }
+  if (removeButton.dataset.removeListenerSet) {
```

Comment:
question - am I right in thinking we are adding in an attribute `removeListenerSet` (set to `true`) to each of the remove buttons? I'm not following, why don't we set it up as an attribute from the beginning instead of searching for all the remove buttons and adding it then?

Reply:
What we are doing is adding this attribute to mark that we've set up the "remove" listener, so that for a given button we don't accidentally set up the listener twice. Otherwise when we click the remove button we might trigger the logic to run twice.

The logic basically says "If I haven't already marked this task as done, do the task and then mark the task as done"